packages/typegpu/src/std/numeric.ts

```diff
@@ -19,13 +19,81 @@ export function sub<T extends vBase>(lhs: T, rhs: T): T {
   return VectorOps.sub[lhs.kind](lhs, rhs);
 }
 
-export function mul<T extends vBase>(s: number, v: T): T {
+export function mul<T extends vBase>(s: number | T, v: T): T {
   if (inGPUMode()) {
     return `(${s} * ${v})` as unknown as T;
   }
-  return VectorOps.mul[v.kind](s, v);
+  if (typeof s === 'number') {
+    return VectorOps.mulSxV[v.kind](s, v);
+  }
+  return VectorOps.mulVxV[v.kind](s, v);
+}
+
+export function abs<T extends vBase | number>(value: T): T {
+  if (inGPUMode()) {
+    return `abs(${value})` as unknown as T;
+  }
+  if (typeof value === 'number') {
+    return Math.abs(value) as T;
+  }
+  return VectorOps.abs[value.kind](value) as T;
+}
+
+/**
+ * @privateRemarks
+ * https://www.w3.org/TR/WGSL/#ceil-builtin
+ */
+export function ceil<T extends vBase | number>(value: T): T {
+  if (inGPUMode()) {
+    return `ceil(${value})` as unknown as T;
+  }
+  if (typeof value === 'number') {
+    return Math.ceil(value) as T;
+  }
+  return VectorOps.ceil[value.kind](value) as T;
+}
+
+/**
+ * @privateRemarks
+ * https://www.w3.org/TR/WGSL/#clamp
+ */
+export function clamp<T extends vBase | number>(value: T, low: T, high: T): T {
+  if (inGPUMode()) {
+    return `clamp(${value}, ${low}, ${high})` as unknown as T;
+  }
+  if (typeof value === 'number') {
+    return Math.min(Math.max(low as number, value), high as number) as T;
+  }
+  return VectorOps.clamp[value.kind](value, low as vBase, high as vBase) as T;
+}
+
+// TODO: Accept vectors into `cos`
+/**
+ * @privateRemarks
+ * https://www.w3.org/TR/WGSL/#cos-builtin
+ */
+export function cos(radians: number): number {
+  if (inGPUMode()) {
+    return `cos(${radians})` as unknown as number;
+  }
+  return Math.cos(radians);
+}
+
+/**
+ * @privateRemarks
+ * https://www.w3.org/TR/WGSL/#cross-builtin
+ */
+export function cross<T extends v3f | v3i | v3u>(a: T, b: T): T {
```

Comment:
```suggestion
export function cross<T extends vBase>(a: T, b: T): T {
```
for consistency

Reply:
It was intentional, as `cross` should only work for vectors with 3 elements.

Reply:
oh, i see